docs/examples/grpc/README.md

```diff
@@ -0,0 +1,147 @@
+# gRPC
+
+This example demonstrates how to route traffic to a gRPC service through the
+nginx controller.
+
+## Prerequisites
+
+1. You have a kubernetes cluster running.
+2. You have a domain name such as `example.com` that is configured to route
+   traffic to the ingress controller.
+3. You have the nginx-ingress controller installed in typical fashion (must be
+   at least <quay.io/kubernetes-ingress-controller/nginx-ingress-controller:0.13.0>) for grpc support.
+4. You have a backend application running a gRPC server and listening for TCP
+   traffic.  If you prefer, you can use the sample
+   [fortune-teller](https://github.com/stackb/grpc-fortune-teller) application provided
+   here as an example. 
+
+### Step 1: kubernetes `Deployment`
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: fortune-teller-app
+  labels:
+    k8s-app: fortune-teller-app
+  namespace: default
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        k8s-app: fortune-teller-app
+    spec:
+      containers:
+      - name: fortune-teller-app
+        image: container.stack.build/pcj/fortune-teller:latest
+        ports:
+        - containerPort: 50051
+          name: grpc
+```
+
+Here we have fairly standard kubernetes deployment object.  It is running a
+grpc service listening on port `50051`.
+
+The sample application
+[fortune-teller-app](https://github.com/stackb/grpc-fortune-teller) is a grpc
+server implemented in go. Here's the stripped-down implementation:
+
+```go
+func main() {
+	grpcServer := grpc.NewServer()
+	fortune.RegisterFortuneTellerServer(grpcServer, &FortuneTeller{})
+	lis, _ := net.Listen("tcp", ":50051")
+	grpcServer.Serve(lis)
+}
+```
+
+The takeaway is that we are not doing any TLS configuration on the server (as we
+are terminating TLS at the ingress level, grpc traffic will arrive unencrypted).
+
+For your own application you may or may not want to do this.  If you prefer to
+forward encrypted traffic to your POD and terminate TLS at the gRPC server
+itself, add the ingress annotation `nginx.ingress.kubernetes.io/secure-backends:
+"true"`.
+
+### Step 2: the kubernetes `Service`
+
+```yaml
```

Comment:
Maybe a link to the corresponding YAML files would be better than copying their content directly here?
Or the other way around, keep the content of the manifests in the markdown but delete the files.

Reply:
Sure I can delete the redundant content.  I prefer the readability of the yaml in the markdown but could be convinced otherwise.  Unless another feels strongly, I'll just delete the yaml files.

Reply:
I like the style in other docs where we provide complete `kubectl` command:
```
cat docs/examples/grpc/svc.yaml | kubectl apply --namespace=fortune-teller-app -f -
```

Reply:
@ElvinEfendi done.